react/src/components/DefaultProviders.tsx

```diff
@@ -205,7 +208,34 @@ const DefaultProvidersForWebComponent: React.FC<DefaultProvidersProps> = ({
 }) => {
   const cache = useMemo(() => createCache(), []);
   const [lang] = useCurrentLanguage();
+
+  const [userCustomThemeConfig] = useBAISettingUserState('custom_theme_config');
+  const [isThemePreviewMode] = useSessionStorageState('isThemePreviewMode', {
+    defaultValue: false,
+  });
   const themeConfig = useCustomThemeConfig();
+  const defaultThemeConfig =
+    isThemePreviewMode && !_.isEmpty(userCustomThemeConfig)
+      ? userCustomThemeConfig
+      : themeConfig;
+
+  const reloadPreviewWindow = useEffectEvent(() => {
+    if (!isThemePreviewMode) return;
+
+    const handleLocalStorageChange = (e: StorageEvent) => {
+      if (e.key === 'backendaiwebui.settings.user.custom_theme_config') {
+        window.location.reload();
+      }
+    };
+    window.addEventListener('storage', handleLocalStorageChange);
+    return () =>
+      window.removeEventListener('storage', handleLocalStorageChange);
+  });
```

Comment:
this uses useEffectEvent which may not be the correct pattern - this hook is meant for events, not effect setup.